### DISCUSSION TITLE
Introduce the concept of "sub-prompt" to improve prompt reusability

### DISCUSSION THREAD

**Discussion by khaelys** [langfuse/langfuse#2039] [Ideas]
### Describe the feature or potential improvement

I have a lot of use cases where prompt sections must be reused in different prompts.  At the moment I just created a prompt for each "section", fetch it first, and pass it as a variable to my prompt.

I'll make an example with two prompts for clarification:

### Sub-prompt
My prompt section

### Prompt 1
{{section}}

Prompt 1 details...

### Prompt 2
{{section}}

Prompt 2 details...

---

This is very useful because I have a central place where I can update all the prompts that use the same section at once, reducing duplication and risk of mistakes (for example I forgot to update the section in one of the prompts that use it).

**Comment by marcklingen** [comment]
Thanks again for sharing this idea! Can you add your plus one to this thread? #1660

Sorry for the slow response here!
